### DISCUSSION TITLE
System.Runtime.InteropServices.COMException (0x88980406): UCEERR_RENDERTHREADFAILURE (0x88980406)

### DISCUSSION THREAD

**Discussion by sam-wheat** [dotnet/wpf#7465] [Q&A]
I am diagnosing the WPF render thread failure (WPF app TargetFramework net6.0-windows).  In accordance with [this documentation related to the issue](https://learn.microsoft.com/en-us/troubleshoot/developer/dotnet/framework/general/wpf-render-thread-failures)  I am trying to gather any diagnostic info I can.  The stacktrace when the app crashes is:

	at System.Windows.Media.Composition.DUCE.Channel.SyncFlush()
	at System.Windows.Interop.HwndTarget.UpdateWindowSettings(Boolean enableRenderTarget, Nullable`1 channelSet)
	at System.Windows.Interop.HwndTarget.UpdateWindowPos(IntPtr lParam)
	at System.Windows.Interop.HwndTarget.HandleMessage(WindowMessage msg, IntPtr wparam, IntPtr lparam)
	at System.Windows.Interop.HwndSource.HwndTargetFilterMessage(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
	at MS.Win32.HwndWrapper.WndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
	at MS.Win32.HwndSubclass.DispatcherCallbackOperation(Object o)
	at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)

	System.Runtime.InteropServices.COMException (0x88980406): UCEERR_RENDERTHREADFAILURE (0x88980406)
	at System.Windows.Media.Composition.DUCE.Channel.SyncFlush()
	at System.Windows.Interop.HwndTarget.UpdateWindowSettings(Boolean enableRenderTarget, Nullable`1 channelSet)
	at System.Windows.Interop.HwndTarget.UpdateWindowPos(IntPtr lParam)
	at System.Windows.Interop.HwndTarget.HandleMessage(WindowMessage msg, IntPtr wparam, IntPtr lparam)
	at System.Windows.Interop.HwndSource.HwndTargetFilterMessage(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
	at MS.Win32.HwndWrapper.WndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
	at MS.Win32.HwndSubclass.DispatcherCallbackOperation(Object o)
	at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
	at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
	at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
   
   
Additional info gathered while running the app under Windbg:

	// snip.. many .dll's loaded will post additional logging on request

	ModLoad: 00007ff9`f33c0000 00007ff9`f33c9000   C:\Program Files\dotnet\shared\Microsoft.NETCore.App\6.0.11\System.Diagnostics.StackTrace.dll
	ModLoad: 00007ff9`fa030000 00007ff9`fa0d2000   C:\windows\SYSTEM32\sxs.dll
	mincore\com\oleaut32\dispatch\ups.cpp(2122)\OLEAUT32.dll!00007FF9FB839DD6: (caller: 00007FF9FB8391E9) ReturnHr(1) tid(67a8) 8002801D Library not registered.
	ModLoad: 00007ff9`c2a50000 00007ff9`c2ab6000   C:\windows\SYSTEM32\OLEACC.dll
	ModLoad: 0000021b`c4930000 0000021b`c4938000   C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App\6.0.11\Accessibility.dll
	ModLoad: 00007ff9`74fd0000 00007ff9`750cf000   C:\Program Files\dotnet\shared\Microsoft.NETCore.App\6.0.11\Microsoft.CSharp.dll
	ModLoad: 00007ff9`f2140000 00007ff9`f2147000   C:\Program Files\dotnet\shared\Microsoft.NETCore.App\6.0.11\System.Runtime.Serialization.Primitives.dll
	MIL FAILURE: Unexpected HRESULT 0x80070057 in caller: batch processing error
	MIL FAILURE: Unexpected HRESULT 0x80070057 in caller: The render thread failed unexpectedly.
	(65d4.67a8): CLR exception - code e0434352 (first chance)
	ModLoad: 00007ff9`73f20000 00007ff9`740e2000   C:\Program Files\dotnet\shared\Microsoft.NETCore.App\6.0.11\Microsoft.DiaSymReader.Native.amd64.dll
	ModLoad: 00007ff9`73e00000 00007ff9`73f14000   C:\Program Files\dotnet\shared\Microsoft.NETCore.App\6.0.11\System.Reflection.Metadata.dll
	(65d4.67a8): CLR exception - code e0434352 (first chance)
	(65d4.67a8): CLR exception - code e0434352 (first chance)
	(65d4.67a8): Unknown exception - code 0000071a (first chance)
	(65d4.67a8): CLR exception - code e0434352 (first chance)
	(65d4.67a8): CLR exception - code e0434352 (first chance)
	(65d4.67a8): CLR exception - code e0434352 (!!! second chance !!!)
	KERNELBASE!RaiseException+0x69:
	00007ff9`fa3fcd29 0f1f440000      nop     dword ptr [rax+rax]

My question relates to the failure to load OLEAUT32.dll  - is this something I sould pursue?  Will I add to my grief if I try to register this dll on my machine?  I have done a fair amount of research on this but most of the related issues are over my head.  This is a production app.  Any feedback or direction will be greatly appreciated.

Thanks

	C:\>dotnet --info
	.NET SDK:
	 Version:   7.0.100
	 Commit:    e12b7af219

	Runtime Environment:
	 OS Name:     Windows
	 OS Version:  10.0.19044
	 OS Platform: Windows
	 RID:         win10-x64
	 Base Path:   C:\Program Files\dotnet\sdk\7.0.100\

	Host:
	  Version:      7.0.0
	  Architecture: x64
	  Commit:       d099f075e4

	.NET SDKs installed:
	  3.1.425 [C:\Program Files\dotnet\sdk]
	  5.0.407 [C:\Program Files\dotnet\sdk]
	  5.0.408 [C:\Program Files\dotnet\sdk]
	  7.0.100 [C:\Program Files\dotnet\sdk]

	.NET runtimes installed:
	  Microsoft.AspNetCore.App 3.1.24 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
	  Microsoft.AspNetCore.App 3.1.31 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
	  Microsoft.AspNetCore.App 5.0.16 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
	  Microsoft.AspNetCore.App 5.0.17 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
	  Microsoft.AspNetCore.App 6.0.11 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
	  Microsoft.AspNetCore.App 7.0.0 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
	  Microsoft.NETCore.App 3.1.24 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
	  Microsoft.NETCore.App 3.1.31 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
	  Microsoft.NETCore.App 5.0.16 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
	  Microsoft.NETCore.App 5.0.17 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
	  Microsoft.NETCore.App 6.0.11 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
	  Microsoft.NETCore.App 7.0.0 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
	  Microsoft.WindowsDesktop.App 3.1.24 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
	  Microsoft.WindowsDesktop.App 3.1.31 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
	  Microsoft.WindowsDesktop.App 5.0.16 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
	  Microsoft.WindowsDesktop.App 5.0.17 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
	  Microsoft.WindowsDesktop.App 6.0.11 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
	  Microsoft.WindowsDesktop.App 7.0.0 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

	Other architectures found:
	  x86   [C:\Program Files (x86)\dotnet]
		registered at [HKLM\SOFTWARE\dotnet\Setup\InstalledVersions\x86\InstallLocation]

	Environment variables:
	  Not set

	global.json file:
	  Not found

	Learn more:
	  https://aka.ms/dotnet/info

	Download .NET:
	  https://aka.ms/dotnet/download

**Comment by NimbusHex** [comment]
I know this isn't helpful, but I have been running into this exception with WPF applications out in the wild for 10 years now, since I first developed WPF apps in .NET 4.0 to now in .NET 4.8.  It mostly seems to happen when users are running the application for long periods of time (5+ days).  Frankly, I think this is something that is unfixable without a major overhaul of the underlying architecture of WPF itself.
